src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50;
+
+      if (herbivore.health <= 0) {
+        herbivore.die();
+      }
```

Comment:
[CODE PERFORMANCE] - It is preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method. The current implementation may not be optimal as it involves calling the `die` method which in turn filters `Animal.alive` every time an animal's health drops to 0 or below. Consider using a single filter operation after all bite actions are processed to clean up dead animals.